source/vstd/std_specs/vec.rs

```diff
@@ -248,6 +248,14 @@ pub fn ex_vec_truncate<T, A: Allocator>(vec: &mut Vec<T, A>, len: usize)
     vec.truncate(len)
 }
 
+#[verifier::external_fn_specification]
+pub fn ex_vec_extend_from_slice<T: Clone, A: Allocator>(vec: &mut Vec<T, A>, slice: &[T])
+    ensures
+        vec@ == old(vec)@ + slice@,
```

Comment:
The new elements in `vec` are generated by calling clone, so I think this needs to be based on the ensures from clone, like `ex_vec_clone` is.